src/core/crypto/secp256k1.ts

```diff
@@ -173,6 +175,26 @@ export class Secp256k1PrivateKey extends PrivateKey {
     const bytes = secp256k1.getPublicKey(this.key.toUint8Array(), false);
     return new Secp256k1PublicKey(bytes);
   }
+
+  /**
+   * Derives a private key from a mnemonic seed phrase.
+   *
+   * @param path the BIP44 path
+   * @param mnemonics the mnemonic seed phrase
+   */
+  static fromDerivationPath(path: string, mnemonics: string): Uint8Array {
+    if (!isValidBIP32Path(path)) {
+      throw new Error(`Invalid derivation path ${path}`);
+    }
+    const { privateKey } = HDKey.fromMasterSeed(mnemonicToSeed(mnemonics)).derive(path);
+
+    // library returns privateKey as Uint8Array | null
+    if (privateKey == null) {
```

Comment:
Unless empty byte array is possible
```suggestion
    if (privateKey === null) {
```